core/src/main/java/org/elasticsearch/index/query/AndQueryBuilder.java

```diff
@@ -42,12 +42,15 @@
 
     public static final String NAME = "and";
 
-    private ArrayList<QueryBuilder> filters = Lists.newArrayList();
+    private final ArrayList<QueryBuilder> filters = Lists.newArrayList();
 
     private String queryName;
 
     static final AndQueryBuilder PROTOTYPE = new AndQueryBuilder();
 
+    /**
+     * @param filters nested filters, no <tt>null</tt> values are allowed
+     */
     public AndQueryBuilder(QueryBuilder... filters) {
         for (QueryBuilder filter : filters) {
             this.filters.add(filter);
```

Comment:
maybe we should go through the add method here to make sure that null values get rejected properly